web-extension/test/node-rpc.spec.js

```diff
@@ -0,0 +1,68 @@
+import NodeRPC from '../backend/node-rpc.js'
```

Comment:
So I actually think we can focus the test approach a bit more here to ensure the test is far less likely to pass through luck if there is ever a change in logic or a bug is introduced. The mix of nodes tests leaves us likely enough to pick a healthy node. I think these test cases below would be a bit more of a focussed test approach:

1. 3 unhealthy nodes and a slow node - check we always get the slow node (if that is the desired outcome???)
2. 3 unhealthy nodes and one healthy node - check we always get the healthy one. 
3. 3 slow nodes and one healthy node - check we always get the healthy one
4. 1 healthy, 1 slow, 1 unhealthy - check we always get healthy

These test cases could potentially replace the 'mix of nodes' test case but I will leave that up to you if you still want it.

To clarify it doesn't have to be 3 of the nodes we do not wants vs 1 of the node we do want. IMO the larger the number of the nodes we do not want and the smaller the number of the nodes we do want makes for a better test, but anything upwards of 3 feels about right